node/import_map.ts

```diff
@@ -28,22 +28,50 @@ class ImportMap {
     })
   }
 
-  static resolve(importMapFile: ImportMapFile) {
+  // Transforms an import map by making any relative paths use a different path
+  // as a base.
+  static resolve(importMapFile: ImportMapFile, rootPath?: string) {
     const { baseURL, ...importMap } = importMapFile
     const parsedImportMap = parse(importMap, baseURL)
+    const { imports = {} } = parsedImportMap
+    const newImports: Record<string, string> = {}
 
-    return parsedImportMap
+    Object.keys(imports).forEach((specifier) => {
+      const url = imports[specifier]
+
+      // If there's no URL, don't even add the specifier to the final imports.
+      if (url === null) {
+        return
+      }
+
+      // If this is a file URL, we might want to transform it to use another
+      // root path, as long as that root path is defined.
+      if (url.protocol === 'file:' && rootPath !== undefined) {
+        // We want to use POSIX paths for the import map regardless of the OS
+        // we're building in.
+        const path = relative(rootPath, fileURLToPath(url)).split(sep).join(posix.sep)
+        const value = isAbsolute(path) ? path : `.${posix.sep}${path}`
+
+        newImports[specifier] = value
+
+        return
+      }
+
+      newImports[specifier] = url.toString()
+    })
+
+    return { ...parsedImportMap, imports: newImports }
   }
 
   add(file: ImportMapFile) {
     this.files.push(file)
   }
 
-  getContents() {
-    let imports: Record<string, URL | null> = {}
+  getContents(rootPath?: string) {
```

Comment:
Shouldn't rootPath be mandatory? For example `. toDataURL()` would now create a different import map?

Reply:
We can't have relative paths when the import map is described as a data URL, so in that situation we'll leave the paths as absolute.